### ISSUE TITLE
Write user code example

### ISSUE THREAD

**Issue by qingyun-wu** [microsoft/FLAML#1072]

**Comment by qingyun-wu** [comment]
```python
# Use feedparser to fetch data from arxiv.org
import feedparser
import pickle

cs_feed = feedparser.parse("http://export.arxiv.org/rss/cs")
stat_feed = feedparser.parse("http://export.arxiv.org/rss/stat")
all_cs_data = []
all_stat_data = []
# Loop through each entry (article)
for entry in cs_feed.entries:
    all_cs_data.append(entry.summary)
for entry in stat_feed.entries:
    all_cs_data.append(entry.summary)


# Create an assistant agent
from flaml.autogen.agent import AssistantAgent, UserProxyAgent
assistant = AssistantAgent(name="summary_assistant")
user = UserProxyAgent(name="user", data=cs_data)
assistant.receive(f"Summarize research trend in CS. I have a list of abstract of latest papers from arxiv. You can request data you can digest.")
# during the interaction, the UserProxyAgent can request data from the user to summarize until all the data
# has been read

# serialize the agents
pickle.dump(assistant, open("assistant.pkl", "wb"))
pickle.dump(user, open("user.pkl", "wb"))

# load the agents
assistant = pickle.load(open("assistant.pkl", "rb"))
user = pickle.load(open("user.pkl", "rb"))

# provide new data to the assistant
user.set_data(all_stat_data)
assistant.receive(f"Summarize research trend in CS and stats. I have a list of abstract of latest papers from arxiv. You can request data you can digest.")



```

**Comment by weilinear** [comment]
Quick question regarding the Scenario 2. 

1. Will the learning_results template/structure being a hyper-parameter, or a learnable parameter? For example, I'm wondering if the learning results on research trends will change during the learning process - seeing a few samples will result in a very fine-grained trend (e.g. a particular method name) and the trend will be more abstract given more data (e.g. a particular field name).
2. Does the assistant need to know if only partial data (and what % of data) is given?
3. Is there any parameter to balance the assistant to learn from current batch of data and not forget about the previously given data?

Thanks! (Context, I was discussing with @sonichi on https://github.com/microsoft/FLAML/issues/1063 for our use case)

**Comment by qingyun-wu** [comment]
> 2\. Does the assistant need to know if only partial data (and what % of data) is given?

@weilinear, thank you for your interest and questions.  Please find my responses to you questions below:

1.  I don't have a definitive answer to this question at the moment. Generally speaking, the nature of `learning_results` is likely to be dependent on the specific learning algorithm employed. It could potentially be a hyper-parameter or a learnable parameter. 
2.  In many cases, it is impossible for the assistant to know if only partial data is given. For example, one can always choose to feed new data to the assistant whenever new data is available. Thus, I think we'd better design the assistant in a way such that it does not need to know if only partial data (and what % of data) is given. 
3.  One potential way to make such a balance: In the `learning_constraints`, we could consider including a `learning_window_size` parameter to help regulate the extent of prior data considered for learning.

**Comment by weilinear** [comment]
Thanks for the explanation. A few follow-up questions

- What is considered a `learning algorithm` in this context?
- How will `learning_rate` works here and what will be the assumption we could make about higher/lower learning rate? Will that be some measurements of how fast `learning_results` changes?

**Comment by qingyun-wu** [comment]
> Thanks for the explanation. A few follow-up questions
> 
> * What is considered a `learning algorithm` in this context?
> * How will `learning_rate` works here and what will be the assumption we could make about higher/lower learning rate? Will that be some measurements of how fast `learning_results` changes?

Hi @weilinear, the `learning algorithm` is a function that can take learning data and learning objective, and optionally previous learning results as inputs and outputs learning results. In the context of a summarization task, a large language model could be used as the learning function.   

The learning rate mentioned in my previous response is just an example of the potential ways for the user to make balance old learning_results and learning from new data. How it works depends on the specific learning function. To avoid confusion, I have removed `learning_rate` from the code example and only keep a field named `learning_func`.  

Thank you!

**Comment by qingyun-wu** [comment]
> 

Hi @weilinear,

Thank you for your interest. Could you please take a look at this PR: https://github.com/microsoft/FLAML/pull/1098
, especially the continual summarization example code in this file: https://github.com/microsoft/FLAML/blob/6891db656de7b8de4ecbaad892abafee8411d3a8/test/autogen/test_continual_summarization.py

Thank you!

**Comment by qingyun-wu** [comment]
Hi @weilinear,
An example use case of continual summarization is demonstrated briefly in this notebook: https://github.com/microsoft/FLAML/blob/67a23b167ecad631882fc9c781a1178dc4b5cf50/notebook/autogen_agent_continual_summarization.ipynb

Could you take a look at this demo and share your comments/suggestions? I'd like to have a chat with you if you have time.

**Comment by weilinear** [comment]
Thanks @qingyun-wu. Let me take a look at the demo and will leave some comments. I'm pretty much booked this week. Let's target to have a chat sometime late next week.
